src/js/controls/record-toggle.js

```diff
@@ -72,12 +72,45 @@ class RecordToggle extends Button {
     handleClick(event) {
         let recorder = this.player_.record();
         if (!recorder.isRecording()) {
-            recorder.start();
+            if (recorder.countdown.length) {
+                // @todo move to recorder
```

Comment:
what's up with the todo?

Reply:
im currently on holiday so please implement and i'll review at some point

Reply:
I tested acrosss all the demosexamples/*.html files and it workds prertty good apart from the image-only.html (which is not launching by `RecordToggle`).


Next I'll add a "countdown" state to the `Recorder` and move everything to the `Recorder.start()` + add a condition around the ` this.createSnapshot();` call to display a countdown.